k0s install does not validate config

Install command  will fail if provided configuration file doesn't pass validation

```
$ ./k0s install controller -c ./val.yaml 
ERRO[2021-02-25 15:34:38] failed to get cluster setup: 
001-k0s-privileged is not a built-in pod security policy 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1ac01b7]

goroutine 1 [running]:
github.com/k0sproject/k0s/pkg/install.CreateControllerUsers(0x0, 0xc0004aa940, 0x1b, 0xc000ea1960, 0x10, 0xc000ea1950, 0x10, 0x228dd6e, 0x12, 0x2275f72, ...)
        /root/go/src/github.com/k0sproject/k0s/pkg/install/users.go:35 +0x37
github.com/k0sproject/k0s/cmd.setup(0x2270a5c, 0xa, 0xc0004345a0, 0x2, 0x2, 0x1, 0x1)
        /root/go/src/github.com/k0sproject/k0s/cmd/install.go:92 +0x225
github.com/k0sproject/k0s/cmd.glob..func8(0x3a0b880, 0xc000434520, 0x0, 0x2, 0x0, 0x0)
        /root/go/src/github.com/k0sproject/k0s/cmd/install.go:60 +0x114
github.com/spf13/cobra.(*Command).execute(0x3a0b880, 0xc0004344e0, 0x2, 0x2, 0x3a0b880, 0xc0004344e0)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0x3a0c840, 0xc00020a030, 0x23c0420, 0xc0013b3f48)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/k0sproject/k0s/cmd.Execute()
        /root/go/src/github.com/k0sproject/k0s/cmd/root.go:152 +0x31
```




**Version**
```
$ k0s version
v0.11.0-beta.2-16-g02cddab
```
**Platform**
Which platform did you run k0s on?
```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.10
Release:        20.10
Codename:       groovy
```

--- a/cmd/install.go
+++ b/cmd/install.go
@@ -59,6 +59,7 @@ With controller subcommand you can setup a single node cluster by running:
 			flagsAndVals = append(flagsAndVals, cmdFlagsToArgs(cmd)...)
 			return setup("controller", flagsAndVals)
 		},
+		PreRunE: preRunValidateConfig,
 	}
 
 	installWorkerCmd = &cobra.Command{
@@ -73,6 +74,7 @@ Windows flags like "--api-server", "--cidr-range" and "--cluster-dns" will be ig
 			flagsAndVals = append(flagsAndVals, cmdFlagsToArgs(cmd)...)
 			return setup("worker", flagsAndVals)
 		},
+		PreRunE: preRunValidateConfig,
 	}
 )
 
@@ -97,6 +99,14 @@ func setup(role string, args []string) error {
 	err := install.EnsureService(args)
 	if err != nil {
 		logrus.Errorf("failed to install k0s service: %v", err)
+	}
+	return nil
+}
+
+func preRunValidateConfig(cmd *cobra.Command, args []string) error {
+	err := validateConfig(cfgFile)
+	if err != nil {
+		return err
 	}
 	return nil
 }

--- a/cmd/install.go
+++ b/cmd/install.go
@@ -104,9 +104,5 @@ func setup(role string, args []string) error {
 }
 
 func preRunValidateConfig(cmd *cobra.Command, args []string) error {
-	err := validateConfig(cfgFile)
-	if err != nil {
-		return err
-	}
-	return nil
+	return validateConfig(cfgFile)
 }

--- a/cmd/reset.go
+++ b/cmd/reset.go
@@ -41,6 +41,7 @@ var (
 			}
 			return reset()
 		},
+		PreRunE: preRunValidateConfig,
 	}
 )
 

